File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r = 0,len(height)-1
        h = max(height)
        max_area = 0
        while l<r:
            width = r - l
            max_area = max(max_area, min(height[l], height[r])*width)

            if height[l]<height[r]:
                l += 1
            else:
                r -= 1
            if (r-l)*h < max_area:
                return max_area

        return max_area

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea=0;
        int j= height.size()-1;
        int i=0;
        while(i <= j){
            int area = abs(i-j)*min(height[i],height[j]);
            maxArea = max(maxArea,area);
            if(height[i]>=height[j]){
                j--;
                continue;
            }
            else{
                i++;
                continue;
            }
        }
        return maxArea;
    }
};